src/Cowsay.zig

```diff
@@ -11,23 +11,17 @@ const default_cow =
     \\    ||     ||
 ;
 
-var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
-
-
-
 /// writer for output. must be an AnyWriter. Use `.any()` to convert to AnyWriter before passing in.
-w: std.io.AnyWriter,
+writer: std.io.AnyWriter,
 /// the eyes. will substitute the first two `o` of the cow
 eyes: [2]u8 = [2]u8{ 'o', 'o' },
 
-/// allocator to use when formatting message
-allocator: std.mem.Allocator = general_purpose_allocator.allocator(),
 /// private variables
 thinking: bool = false,
-max_line_length: u32 = 0,
-offset: u32 = 0,
+max_line_length: usize = 0,
+offset: usize = 0,
+cow: []const u8 = &.{},
```

Comment:
Forgot to leave a comment here; it's preferable to use a slice here rather than manually keeping track of buffer length. This will make it more convenient to work with standard library functions that take a slice, generally make other slicing operations/accesses more concise, etc.